beacon_node/operation_pool/src/lib.rs

```diff
@@ -96,6 +97,43 @@ impl<T: EthSpec> OperationPool<T> {
         self.attestations.read().values().map(Vec::len).sum()
     }
 
+    /// Return all valid attestations for the given epoch, for use in max cover.
+    fn get_valid_attestations_for_epoch<'a>(
+        &'a self,
+        epoch: Epoch,
+        all_attestations: &'a HashMap<AttestationId, Vec<Attestation<T>>>,
+        state: &'a BeaconState<T>,
+        total_active_balance: u64,
+        validity_filter: impl FnMut(&&Attestation<T>) -> bool + Send,
+        spec: &'a ChainSpec,
+    ) -> impl Iterator<Item = AttMaxCover<'a, T>> + Send {
+        let domain_bytes = AttestationId::compute_domain_bytes(
+            epoch,
+            &state.fork,
+            state.genesis_validators_root,
+            spec,
+        );
+        all_attestations
+            .iter()
+            .filter(move |(key, _)| key.domain_bytes_match(&domain_bytes))
+            .flat_map(|(_, attestations)| attestations)
+            .filter(move |attestation| {
+                if attestation.data.target.epoch != epoch {
+                    return false;
```

Comment:
Out of interest, any reason this isn't it's own `filter` statement?

Reply:
No good reason! Just haste :sweat_smile: